### PR TITLE
#BACK-1162 add Carb Unit field to calculator object

### PR DESCRIPTION
**Pull request by jamesraby**
Add carb unit field so UI can calculate carbs correctly

### CODE REVIEW

File: data/types/calculator/calculator_test.go [hunk 1]
```diff
@@ -1622,6 +1625,13 @@ var _ = Describe("Calculator", func() {
 					pointer.FromString("mg/dl"),
 					func(datum *calculator.Calculator, units *string) { datum.BolusID = nil },
 				),
+				Entry("Invalid Carb Units",
+					pointer.FromString("mg/dl"),
```
Comment (pazaan) [inline data/types/calculator/calculator_test.go]:
```suggestion
```
Line not required

File: data/types/calculator/calculator_test.go [hunk 2]
```diff
@@ -1622,6 +1625,13 @@ var _ = Describe("Calculator", func() {
...
+					func(datum *calculator.Calculator, units *string) {
+						datum.CarbUnits = pointer.FromString("Invalid")
+					},
+					errorsTest.WithPointerSourceAndMeta(structureValidator.ErrorValueStringNotOneOf("Invalid", calculator.CarbUnits()), "/carbUnits", NewMeta()),
+				),
```
Comment (pazaan) [inline data/types/calculator/calculator_test.go]: Also add tests for the 2 valid values.

File: data/types/calculator/calculator_test.go (hunk 1 above)
Comment (jamesraby) [reply data/types/calculator/calculator_test.go]: This is confusing - but I believe what I have is correct.  This is just creating a unit parameter.  This is actually not used in this test case - but necessary.  We are setting the `CarbUnit` parameter to invalid in the func - so the test should work fine.

File: data/types/calculator/calculator_test.go (hunk 2 above)
Comment (jamesraby) [reply data/types/calculator/calculator_test.go]: In terms of valid values - it does create valid values (albeit random) in the NewCalculator routine - and that is tested in the first unit test.  We could create one for each valid value - but I do not think that is generally done through the framework.

**Pull request merged**